src/main/webapp/app/shared/user-settings/user-settings-prototype/user-settings-prototype.component.ts

```diff
@@ -0,0 +1,107 @@
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { JhiAlertService } from 'ng-jhipster';
+import { User } from 'app/core/user/user.model';
+import { UserSettingsCategory } from 'app/shared/constants/user-settings.constants';
+import { OptionCore, UserSettings } from 'app/shared/user-settings/user-settings.model';
+import { UserSettingsService } from 'app/shared/user-settings/user-settings.service';
+
+/**
+ * Is used as the abstract user-settings "parent" component with all the necessary basic logic for other "child" components to implement/inherit from.
+ * The @Component decorated and templateUrl are needed to be able to use OnInit
+ */
+@Component({
+    templateUrl: 'user-settings-prototype.component.html',
```

Comment:
It's a good idea to provide this base class that concrete instances of the user settings, such as the notification-settings can inherit from.
As far as I know, this is nowadays achieved by using the `@Directive()` decorator, I've read this [here](https://stackoverflow.com/questions/47882247/angular4-components-inheritance-with-abstract-class/47883806#47883806). I also used for metis postings [here](https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/shared/metis/posting.directive.ts)

Reply:
Thank you very much for this information. @Directives are completely new to me and I tried it out and it works like a charm. 🎉

Reply:
happy to hear, I also applied it for the first time 😄